plugins/ad9081.c

```diff
@@ -0,0 +1,618 @@
+/**
+ * AD9081 (MxFE) Plugin
+ *
+ * Copyright (C) 2019 Analog Devices, Inc.
+ *
+ * Licensed under the GPL-2.
+ *
+ **/
+#include <errno.h>
+#include <stdio.h>
+#include <gtk/gtk.h>
+#include <glib.h>
+
+#include "../osc.h"
+#include "../osc_plugin.h"
+#include "../iio_widget.h"
+#include "../iio_utils.h"
+#include "../config.h"
+#include "dac_data_manager.h"
+
+#define THIS_DRIVER			"AD9081"
+#define AD9081				"axi-ad9081-rx"
+#define DAC_DEVICE			"axi-ad9081-tx"
+#define NUM_MAX_CHANNEL			8
+#define AD9081_MAX_ADC_FREQ_MHZ		4000
+#define AD9081_MAX_DAC_FREQ_MHZ		12000
+/*
+ * 81 = 2 widgets per maximmum of 8 rx channels + 8 widgets per maximun 8 tx
+ * channels + 1 combo box for global RX configuration...
+ */
+#define NUM_MAX_WIDGETS			162
+
+const gdouble mhz_scale = 1000000.0;
+const gdouble k_scale = 1000.0;
+
+const char *dac_name;
```

Comment:
I'm not sure that dac_name should be set global. Is this plugin going to be instanced more than once? If so, all instances will return the same pointer through call get_dac_dev_names().
I know this was implemented in a separated PR but I missed this issue.

Reply:
The plugin is going to be instantiated multiple times. So this must not be global.
@dbogdan added this only very recently.

Practically only the instance who has a RX buffer will also have the DAC buffer.
So for the Quad MxFE use case this shouldn't matter. But if you have multiple independent instances this will lead to an issue.

Reply:
I'll move the dac_name to plugin_private and I'll pass osc_plugin to get_dac_dev_names(). I'll create a patch right now.

Reply:
See: https://github.com/analogdevicesinc/iio-oscilloscope/pull/236 and https://github.com/analogdevicesinc/iio-oscilloscope/pull/237.